src/Sylius/Component/Product/spec/Model/ProductSpec.php

```diff
@@ -176,10 +176,9 @@ function its_master_variant_should_be_mutable_and_define_given_variant_as_master
     function it_should_not_add_master_variant_twice_to_collection(VariantInterface $variant)
     {
         $variant->isMaster()->willReturn(true);
-        $variant->isDeleted()->willReturn(false);
 
-        $variant->setProduct($this)->shouldBeCalled();
-        $variant->setMaster(true)->shouldBeCalled();
+        $variant->setProduct($this)->shouldBeCalledTimes(1);
+        $variant->setMaster(true)->shouldBeCalledTimes(2);
```

Comment:
Actually it is not anything related to `it_should_not_add_master_variant_twice_to_collection` and will make that spec failing, even if that sentence is still true.

Btw. this spec is invalid, because `Product::hasVariants()` calls `Product::getVariants()` which returns all variants but not the master one(s). I can add two variants and they can be both in `Product::$variants` but the spec will still pass.